cpp/src/io/orc/reader_impl.cu

```diff
@@ -319,7 +319,9 @@ rmm::device_buffer reader::impl::decompress_stripe_data(
   }
   CUDF_EXPECTS(total_decomp_size > 0, "No decompressible data found");
 
-  rmm::device_buffer decomp_data(total_decomp_size, stream);
+  // Buffer needs to be padded.
+  rmm::device_buffer decomp_data(
+    cudf::util::round_up_safe(total_decomp_size, BUFFER_PADDING_MULTIPLE), stream);
```

Comment:
output of (presumed) nvcomp snappy decompression.

Reply:
Issue from `gpuDecodeOrcColumnData`.